src/components/Card.js

```diff
@@ -1,62 +1,90 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleCardImage) {
+  constructor( {
+    name, 
+    link,  
+    isLiked,
+    _id,
+  },
+    cardSelector,
+    handleCardImage,
+    handleDelete,
+    userId,
+    handleCardLike,
+
+  ) {
     this._name = name;
     this._link = link;
+    this._isLiked = isLiked;
+    this._cardId = _id;
     this._cardSelector = cardSelector;
     this._handleCardImage = handleCardImage;
+    this._handleDeleteCard = handleDelete;
+    this._userId = userId;
+    this._handleCardLike = handleCardLike
+
+  }
+
+  getView() {
+    this._cardElement = this._getTemplate();
+    this._handleCardDeleteButton = this._cardElement.querySelector(".card__delete-button");
+    this._cardLikeButton = this._cardElement.querySelector(".card__like-button");
+    this._setEventListeners();
+    this.renderLikes();
+    this._cardElement.querySelector(".card__like-button");
+    this._cardElement.querySelector(".card__image").src = this._link;
+    this._cardElement.querySelector(".card__image").alt = this._name;

```

Comment:
You find the image 3 times in the code. Please, define `this._cardImage =...` for it

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.